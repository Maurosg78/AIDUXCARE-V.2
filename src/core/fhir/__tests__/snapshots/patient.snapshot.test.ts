// src/core/fhir/__tests__/snapshots/patient.snapshot.test.ts
import { describe, it, expect } from 'vitest';
import { internalPatientMock } from '../__fixtures__/internal.mock';
// Importar desde el index principal
import { toFhir } from '../../index';

describe('FHIR Patient snapshot', () => {
  it('matches expected snapshot for a representative internal patient', () => {
    const fhirPatient = toFhir(internalPatientMock, { profile: 'US_CORE' });
    // Validación básica previa al snapshot
    expect(fhirPatient.resourceType).toBe('Patient');
    expect(fhirPatient).toMatchSnapshot();
  });
});
