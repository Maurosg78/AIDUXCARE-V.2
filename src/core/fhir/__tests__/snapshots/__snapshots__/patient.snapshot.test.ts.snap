// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FHIR Patient snapshot > matches expected snapshot for a representative internal patient 1`] = `
{
  "active": true,
  "address": [
    {
      "city": "Valencia",
      "country": "US",
      "line": [
        undefined,
      ],
      "postalCode": undefined,
      "state": "VC",
      "type": "physical",
      "use": "home",
    },
  ],
  "birthDate": "1988-03-15",
  "gender": "female",
  "id": "pt-0001",
  "meta": {
    "lastUpdated": "2025-08-11T20:31:27.760Z",
    "profile": [
      "http://hl7.org/fhir/us/core/StructureDefinition/us-core-patient",
    ],
  },
  "name": [
    {
      "family": "Garc√≠a",
      "given": [
        "Elena",
      ],
      "use": "official",
    },
  ],
  "resourceType": "Patient",
}
`;
